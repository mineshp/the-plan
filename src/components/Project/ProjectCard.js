import React from 'react';
import PropTypes from 'prop-types';
import { Button, Card } from 'semantic-ui-react';
import { formatDate } from '../../helpers/validators/common';

const ProjectCard = ({ data, onDeleteHandler }) => (
    <Card color={data.colour}>
        <Card.Content>
            <Card.Header className="firstletter-capitalise">
                {data.projectName}
                <Button
                    floated="right"
                    icon="list layout"
                    color="blue"
                    href={`/project/${data.projectName}/lists`} // eslint-disable-line no-underscore-dangle
                    title={`View all lists for project ${data.projectName}`}
                />
            </Card.Header>
        </Card.Content>
        <Card.Content>
            <Card.Meta>
                Created by: {data.owner}
            </Card.Meta>
            <Card.Meta>
                {formatDate(data.createdDate)}
            </Card.Meta>
            <Card.Description>
                {data.projectDescription}
            </Card.Description>
        </Card.Content>
        <Card.Content extra>
            <div className="buttons">
                <Button
                    as="a"
                    content="Update"
                    icon="edit"
                    labelPosition="left"
                    color="green"
                    href={`/project/update/${data._id}`} // eslint-disable-line no-underscore-dangle
                />
                <Button
                    content="Delete"
                    icon="trash"
                    labelPosition="left"
                    color="pink"
                    value={data._id} // eslint-disable-line no-underscore-dangle
                    onClick={onDeleteHandler}
                />
            </div>
        </Card.Content>
    </Card>
);

ProjectCard.propTypes = {
    data: PropTypes.shape({
        _id: PropTypes.string.isRequired,
        projectName: PropTypes.string.isRequired,
        colour: PropTypes.string
    }).isRequired,
    onDeleteHandler: PropTypes.func.isRequired

};

export default ProjectCard;
