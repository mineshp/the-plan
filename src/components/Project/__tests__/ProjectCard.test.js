import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import ProjectCard from '../ProjectCard';
import { mockProjectData } from '../../../helpers/test/testData/projectData';

const deleteHandler = jest.fn();

describe('ProjectCard', () => {
    it('renders correctly', () => {
        const project = mockProjectData();

        const tree = renderer.create(
            <MemoryRouter>
                <ProjectCard
                    data={project}
                    key={project._id} // eslint-disable-line no-underscore-dangle
                    onDeleteHandler={deleteHandler}
                />
            </MemoryRouter>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
