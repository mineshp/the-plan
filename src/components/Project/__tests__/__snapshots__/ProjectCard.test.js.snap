// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectCard renders correctly 1`] = `
<div
  className="ui card"
  href={undefined}
  onClick={[Function]}
>
  <div
    className="content"
  >
    <div
      className="header"
    >
      test-project
    </div>
  </div>
  <div
    className="content"
  >
    <div
      className="meta"
    />
    <div
      className="description"
    >
      Add a description here
    </div>
  </div>
  <div
    className="extra content"
  >
    <div
      className="buttons"
    >
      <button
        className="ui olive icon left labeled button"
        disabled={undefined}
        onClick={[Function]}
        role="button"
        tabIndex={undefined}
      >
        <i
          aria-hidden="true"
          className="edit icon"
        />
        Update
      </button>
      <button
        className="ui red icon left labeled button"
        disabled={undefined}
        onClick={[Function]}
        role="button"
        tabIndex={undefined}
        value="12345"
      >
        <i
          aria-hidden="true"
          className="trash icon"
        />
        Delete
      </button>
    </div>
  </div>
</div>
`;
